### PR TITLE
Add DlsClient constructors that accept Config instances

### PR DESCRIPTION
**Pull request by nemanja-boric-sociomantic**
Blocked on #17

### CODE REVIEW

**Comment by nemanja-boric-sociomantic** [comment]
Rebased and unblocked.

File: src/dlsproto/client/DlsClient.d [hunk 1]
```diff
@@ -213,7 +217,9 @@ public class ExtensibleDlsClient ( Plugins ... ) : DlsClient
 
     /***************************************************************************
 
-        Constructor
+        Constructor with support for only the legacy protocol. The constructor
```
Comment (gavin-norman-sociomantic) [inline src/dlsproto/client/DlsClient.d]: "This constructor"

File: src/dlsproto/client/DlsClient.d [hunk 2]
```diff
@@ -320,7 +330,9 @@ public class SchedulingDlsClient : ExtensibleDlsClient!(RequestScheduler)
...
```
Comment (gavin-norman-sociomantic) [inline src/dlsproto/client/DlsClient.d]: "This constructor"

**Review by gavin-norman-sociomantic — APPROVED** [review]

**Comment by nemanja-boric-sociomantic** [comment]
Updated.

**Pull request merged**